cli/provider/cmd.go

```diff
@@ -302,9 +303,9 @@ func (c CmdConfigurator) ValidateFlags(ctx context.Context, cmd *cobra.Command)
 			}
 			return errors.New("missing required -c flag or appCmd in config file")
 		}
-
-		defer utils.GenerateGithubActions(c.logger, c.cfg.Command)
-
+		if cmd.Name() == "record" && c.cfg.GenerateGithubActions {
```

Comment:
I think it should be independent of the command, if the user has recorded test cases from an older version of keploy and now upgraded the version but directly using the test mode then in that case also keploy should generate the GitHub action.